server/src/main/java/org/opensearch/repositories/blobstore/BlobStoreFormat.java

```diff
@@ -0,0 +1,96 @@
+ * serialise the data in Json format and store it on remote store as is. This does not support compression yet (this
+ * can be changed as required). In comparison to {@link ChecksumBlobStoreFormat}, this format does not add any additional
+ * metadata (like header and footer) to the content. Hence, this format does not depend on {@code CodecUtil} from
+ * Lucene library.
+ *
+ * @opensearch.internal
+ */
+public class BlobStoreFormat<T extends ToXContent> {
+
+    private final String blobNameFormat;
+
+    /**
+     * @param blobNameFormat format of the blobname in {@link String#format} format
+     */
+    public BlobStoreFormat(String blobNameFormat) {
+        this.blobNameFormat = blobNameFormat;
+    }
+
+    private String blobName(String name) {
+        return String.format(Locale.ROOT, blobNameFormat, name);
+    }
+
+    private BytesReference serialize(final T obj) throws IOException {
+        try (XContentBuilder xContentBuilder = XContentFactory.jsonBuilder()) {
+            xContentBuilder.startObject();
+            obj.toXContent(xContentBuilder, ToXContent.EMPTY_PARAMS);
+            xContentBuilder.endObject();
+            return BytesReference.bytes(xContentBuilder);
+        }
+    }
+
+    public void writeAsyncWithUrgentPriority(T obj, BlobContainer blobContainer, String name, ActionListener<Void> listener)
+        throws IOException {
+        if (blobContainer instanceof AsyncMultiStreamBlobContainer == false) {
+            write(obj, blobContainer, name);
+            listener.onResponse(null);
+            return;
+        }
+        String blobName = blobName(name);
+        BytesReference bytes = serialize(obj);
+        String resourceDescription = "BlobStoreFormat.writeAsyncWithPriority(blob=\"" + blobName + "\")";
+        try (IndexInput input = new ByteArrayIndexInput(resourceDescription, BytesReference.toBytes(bytes))) {
+            try (
+                RemoteTransferContainer remoteTransferContainer = new RemoteTransferContainer(
+                    blobName,
+                    blobName,
+                    bytes.length(),
+                    true,
+                    WritePriority.URGENT,
+                    (size, position) -> new OffsetRangeIndexInputStream(input, size, position),
+                    null,
+                    ((AsyncMultiStreamBlobContainer) blobContainer).remoteIntegrityCheckSupported()
+                )
+            ) {
+                ((AsyncMultiStreamBlobContainer) blobContainer).asyncBlobUpload(remoteTransferContainer.createWriteContext(), listener);
+            }
+        }
+    }
+
+    private void write(final T obj, final BlobContainer blobContainer, final String name) throws IOException {
+        String blobName = blobName(name);
+        BytesReference bytes = serialize(obj);
+        blobContainer.writeBlob(blobName, bytes.streamInput(), bytes.length(), false);
```

Comment:
The same code exists in `ChecksumBlobStoreFormat`. Any way to have the common code in abstract class?

Reply:
Both the classes have quite a lot different set of arguments and behaviour of the serialise method. If we get more commonality, we can think about common code in abstract class.

Reply:
+1 to check if there is possibility to extract out any common code

Reply:
Made the change as suggested.